internal/workload/podman/podman.go

```diff
@@ -316,3 +325,96 @@ func (p *podman) GenerateSystemdService(podId string, monitoringInterval uint) (
 
 	return svc, nil
 }
+
+// Retrieve all pods logs and send that to the given io.Writer
+func (p *podman) Logs(podID string, res io.Writer) (context.CancelFunc, error) {
+
+	podInfo, err := pods.Inspect(p.podmanConnection, podID, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithCancel(p.podmanConnection)
+	writers := map[string]*bufio.Reader{}
+	for _, container := range podInfo.Containers {
+		res := &bytes.Buffer{}
+		err := p.containerLog(ctx, container.ID, res)
+
+		if err != nil {
+			cancel()
+			return nil, err
+		}
+		writers[container.ID] = bufio.NewReader(res)
+	}
+
+	go func() {
+		for {
+			for containerID, buffer := range writers {
+				resultLine := []byte{}
+				for {
+					line, isPrefix, _ := buffer.ReadLine()
+					if len(line) == 0 {
+						break
+					}
+
+					if !isPrefix {
+						resultLine = append(resultLine, line...)
+					}
+				}
+				if len(resultLine) > 0 {
+					_, err := res.Write([]byte(fmt.Sprintf("%s: %s\n", containerID, resultLine)))
+					if err != nil {
+						log.Errorf("Cannot write container log line: %v", err)
+					}
+				}
+			}
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				continue
+			}
+		}
+	}()
+	return cancel, nil
+}
+
+func (p *podman) containerLog(ctx context.Context, containerID string, res io.Writer) error {
+	opts := containers.LogOptions{
+		Follow: &boolTrue,
+		Stderr: &boolTrue,
+		Stdout: &boolTrue,
+		Tail:   &logTailLines,
+	}
+	stdoutCh := make(chan string, 100)
```

Comment:
why buffered channel?  any why 100 if this is per-container?

Reply:
wanted to have by pod, but we need to have per container in case one pod has multiple containers. About buffer, to not block anything if pod logs are faster than pod reads.